sphinx_gallery/backreferences.py

```diff
@@ -67,9 +68,18 @@ def get_mapping(self):
             if local_name in self.imported_names:
                 # Join import path to relative path
                 full_name = self.imported_names[local_name] + remainder
-                yield name, full_name, class_attr
+                if local_name in self.global_variables:
+                    obj = self.global_variables[local_name]
+                    if remainder:
+                        for level in remainder[1:].split('.'):
+                            obj = getattr(obj, level)
```

Comment:
In some case doing `c = a.b.c` did not work but doing `c = a.b; c = c.c` did. Can't remember offhand if it had to do with module nesting or class attributes or what... 

...but fortunately I was so annoyed by it that I wrote a test to capture it, and if I just do `obj = getattr(obj, remainder[1:])` (what I think you're suggesting) I get:
```
AttributeError: module 'numpy' has no attribute 'random.RandomState'
```
Which is probably because of this somewhat unexpected behavior:
```
>>> import numpy
>>> getattr(numpy, 'random.RandomState')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/larsoner/python/numpy/numpy/__init__.py", line 219, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'random.RandomState'
>>> numpy.random.RandomState
<class 'numpy.random.mtrand.RandomState'>
```

Reply:
I think I am suggesting `obj = remainder[1:].split('.')[-1]` - the last loop of the for loop?

Reply:
Then `obj` is a `str` rather than the actual object

Reply:
Whoops

```
level = remainder[1:].split('.')[-1]
obj = getattr(obj, level)
```

Reply:
Think about how this would work for the NumPy example -- it would try just to do `getattr('numpy', 'RandomState')` in the case above. And this is what the tests show when dropping your code in for the `for` loop:
```
AttributeError: module 'numpy' has no attribute 'RandomState'
```
It shouldn't just use the last thing following the `.` but traverse all levels in sequence, which is what the `for` loop is designed to do.

Reply:
Ah I get it now. Sorry and thanks for explaining.

Reply:
No problem, hopefully it's not too bad to add your changes from #584 on top once this is in -- then it will be my turn to be confused during review :)

Reply:
@larsoner FYI in case you are interested: https://stackoverflow.com/questions/11975781/why-does-getattr-not-support-consecutive-attribute-retrievals

Reply:
Apparently you are allowed to have `.` in attribute names!

Reply:
I hope never to encounter such code :)